src/cppcache/src/ThinClientRegion.hpp

```diff
@@ -416,14 +416,21 @@ class ChunkedFunctionExecutionResponse : public TcrChunkedResult {
 
  public:
   inline ChunkedFunctionExecutionResponse(
-      TcrMessage& msg, bool getResult, ResultCollectorPtr rc,
-      ACE_Recursive_Thread_Mutex* resultCollectorLock = NULL)
+      TcrMessage& msg, bool getResult, ResultCollectorPtr rc)
       : TcrChunkedResult(),
         m_msg(msg),
         m_getResult(getResult),
-        m_rc(rc),
-        m_resultCollectorLock(resultCollectorLock) {}
-
+        m_rc(rc) {}
+
+    inline ChunkedFunctionExecutionResponse(
+                                            TcrMessage& msg, bool getResult, ResultCollectorPtr rc,
+                                            std::shared_ptr<ACE_Recursive_Thread_Mutex> resultCollectorLock)
+    : TcrChunkedResult(),
+    m_msg(msg),
+    m_getResult(getResult),
+    m_rc(rc),
+    m_resultCollectorLock(resultCollectorLock) {}
+    
```

Comment:
fun fact: line 433 has dangling white space that the Github diff tool is choosing to ignore